api/prometheus/v1/api.go

```diff
@@ -1088,20 +1091,35 @@ func WithTimeout(timeout time.Duration) Option {
 	}
 }
 
-func (h *httpAPI) Query(ctx context.Context, query string, ts time.Time, opts ...Option) (model.Value, Warnings, error) {
-	u := h.client.URL(epQuery, nil)
-	q := u.Query()
+// WithLimit can be used to provide an optional maximum number of returned entries for APIs that support that.
+// https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries
+func WithLimit(limit uint64) Option {
+	return func(o *apiOptions) {
+		o.limit = limit
+	}
+}
 
+func formatOptions(q url.Values, opts []Option) url.Values {
```

Comment:
```suggestion
func addOptionalURLParams(q url.Values, opts []Option) url.Values {
```

Reply:
Perhaps more explicit?

Reply:
Arh... [one of the hardest problems in  Computer Science - naming things](https://martinfowler.com/bliki/TwoHardThings.html).